fs-storage/src/file_storage.rs

```diff
@@ -110,32 +119,50 @@ where
     }
 
     fn read_fs(&mut self) -> Result<BTreeMap<K, V>> {
-        let file = fs::File::open(&self.path)?;
-        let reader = BufReader::new(file);
-        let mut lines = reader.lines();
+        if !self.path.exists() {
+            return Err(ArklibError::Storage(
+                self.label.clone(),
+                "File does not exist".to_owned(),
+            ));
+        }
 
-        let new_timestamp = fs::metadata(&self.path)?.modified()?;
-        match lines.next() {
-            Some(header) => {
-                let header = header?;
-                self.verify_version(&header)?;
-                let mut data = String::new();
-                for line in lines {
-                    let line = line?;
-                    if line.is_empty() {
-                        continue;
-                    }
-                    data.push_str(&line);
-                }
-                let data: BTreeMap<K, V> = serde_json::from_str(&data)?;
-                self.timestamp = new_timestamp;
-                Ok(data)
+        // Attempt to parse the file using the legacy version 2 storage format of FileStorage.
```

Comment:
Can we scan version number first without actually parsing the data field? And then parse the data according to what the version number is?

Reply:
Something like this:
1. Read the first line.
2. If it's `version 2`, then call `read_version_2_fs`. If it fails, we don't attempt parsing with v3 syntax.
3. If the first line is not `version 2`, then we attempt parsing with v3 syntax.

Reply:
> Can we scan version number first without actually parsing the data field?

Sure that makes more sense. Modified the code to fix this issue.